py/__init__.pyi

```diff
@@ -0,0 +1,8 @@
+from py._error import ErrorMaker
+
+from . import path
```

Comment:
Seems better to me not to import here, but instead handle as a sub-package in the stubs. That is, add `py/path/__init__.pyi`. (We might want to separate the stubs to some separate source tree, if that's even possible, so it doesn't affect the real package).

Reply:
> We might want to separate the stubs to some separate source tree, if that's even possible, so it doesn't affect the real package

`py/path` is empty without the stubs.
Do you mean to keep `py/path/*.pyi` there then, or do you mean to use something like `py/_typed/…`?

The idea of using the main `__init__.pyi` was to see what's really exposed/used - but I guess since it is used from/via `py/path` that is something mypy looks at also already.

Reply:
I saw later that you already added `py/path` directory, so you can just delete this line already, as it's redundant with the subpackage.

My comment about "separate source tree" was that adding actual `py/path` directory might have some affect on runtime module resolution? With namespace packages and such? I.e. currently `import py.path.foo` goes thru the py import hackery to the private `py._path` package, but if you we add actual `py/path` directory does it break anything?

If not then great :)

Reply:
Not sure, but I assume it is fine, since there is only `.pyi`, no `.py`.